marlowe-test/src/Spec/Marlowe/Semantics/Arbitrary.hs

```diff
@@ -1249,3 +1258,76 @@ instance Arbitrary TransactionOutput where
         TransactionOutput <$> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary
       , Error <$> arbitrary
       ]
+
+
+instance Arbitrary SafetyError where
+  arbitrary =
+    oneof
+      [
+        pure MissingRolesCurrency
+      , pure ContractHasNoRoles
+      , MissingRoleToken <$> arbitrary
+      , ExtraRoleToken <$> arbitrary
+      , RoleNameTooLong <$> arbitrary
+      , InvalidCurrencySymbol <$> arbitrary
+      , TokenNameTooLong <$> arbitrary
+      , InvalidToken <$> arbitrary
+      , NonPositiveBalance <$> arbitrary <*> arbitrary
+      , DuplicateAccount <$> arbitrary <*> arbitrary
+      , DuplicateChoice <$> arbitrary
+      , DuplicateBoundValue <$> arbitrary
```

Comment:
FYI @bwbush : 

1. `DuplicateChoice, DuplicateBoundValue` are things that `next` handles... `next` doesn't propose invalid applies... 
2. For Deposit, `next` also handle Identical Evaluated Deposits... something we can't really check at "compile" time,  is 2 deposits identical in a `when` regardless they value could be consider as a warning ?

Reply:
These `Duplicate...` warnings apply to the state of the contract, not to actions like deposits.